src/signature.h

```diff
@@ -0,0 +1,26 @@
+#ifndef __SIGNATURE_H__
+#define __SIGNATURE_H__
+
+#include <luajit.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <sys/types.h>
+
+/**
+ * \struct MemoryRegion
+ *
+ * Defines a process's memory region
+ */
+typedef struct {
+    uintptr_t start; /*!< The start address of the memory region */
+    uintptr_t end; /*!< The end address of the memory region */
+} MemoryRegion;
```

Comment:
I believe theres already a struct for memory maps, you can expand on that one instead of making a new one just for this (also changing uint64_t to uintptr_t on it)